azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -60,7 +70,7 @@ def __init__(self, endpoint, container_id, role_config_name):
         self.api_versions = None
         self.endpoint = endpoint
         self.container_id = container_id
-        self.deployment_id = None
+        self.deployment_id = self._extract_deployment_id(role_config_name)
```

Comment:
I wonder if we should make the deployment ID as an parameter of `AgentGlobal` class to ensure that its always uptodate. Or is this ID dependent on each GS? Any idea?

Reply:
Deployment ID is a part of each goal state, but I'm not sure when/how it changes. See discussion below about adding AgentGlobals as a dependency on this API.

Reply:
Got it. Thanks for explaining it! :)